practices/optimizations.md

```diff
@@ -0,0 +1,55 @@
+---
+slug: optimizations
+stages:
+  - development
+short_description: Optimization is the process of modifying a software system to make some aspect of it work more efficiently or use fewer resources
+tags:
+  - optimizations
+  - code coverage
+keywords:
+  - code  - quality
+  - merge
+  - testing
+  - codebase
+---
+
+# Optimizations
+
+**TL;DR**
+
+Optimization is an act, process, or methodology of making something (such as a design, system, or decision) as fully perfect, functional, or effective as possible
+
+## What Is an Optimization
+The code optimization in the synthesis phase is a program transformation technique, which tries to improve the intermediate code by making it consume fewer resources (i.e. CPU, Memory) so that faster-running machine code will result.
+
+## Why You Might Want the Optimization
+
+Optimizing an algorithm is beyond the scope of the code optimization phase. So the program is optimized. And it may involve reducing the size of the code. So optimization helps to:
+
+- Reduce the space consumed and increases the speed of compilation.
```

Comment:
Please, write this with your own words. It seems too similar to: _Why Optimize?
Optimizing an algorithm is beyond the scope of the code optimization phase. So the program is optimized. And it may involve reducing the size of the code. So optimization helps to:_ 

_Reduce the space consumed and increases the speed of compilation.
Manually analyzing datasets involves a lot of time. Hence we make use of software like Tableau for data analysis. Similarly manually performing the optimization is also tedious and is better done using a code optimizer.
An optimized code often promotes re-usability._ from https://www.geeksforgeeks.org/code-optimization-in-compiler-design/

Reply:
done